datastore/client.go

```diff
@@ -73,6 +73,17 @@ func (dc *datastoreClient) RunQuery(ctx context.Context, in *pb.RunQueryRequest,
 	return res, err
 }
 
+func (dc *datastoreClient) RunAggregationQuery(ctx context.Context, in *pb.RunAggregationQueryRequest, opts ...grpc.CallOption) (res *pb.RunAggregationQueryResponse, err error) {
+	ctx = trace.StartSpan(ctx, "cloud.google.com/go/datastore.datastoreClient.RunAggregationQuery")
+	defer func() { trace.EndSpan(ctx, err) }()
+
+	err = dc.invoke(ctx, func(ctx context.Context) error {
+		res, err = dc.c.RunAggregationQuery(ctx, in, opts...)
+		return err
+	})
+	return res, nil
```

Comment:
This code never returns an error, I think this shall be: `return res, err`

Reply:
For anyone facing the same issue: this is especially annoying when you use testcontainers for firestore in datastore mode that currently not support aggregated queries and get a nil panic instead of the proper error that the function is not (yet) implemented.

Reply:
Created https://github.com/googleapis/google-cloud-go/pull/8222 to return error